### ISSUE TITLE
#lecture19 Error while fetching settings 

### ISSUE THREAD

**Issue by sami313** [thedigicraft/Atom.CMS#112]
Warning: mysqli_fetch_assoc() expects parameter 1 to be mysqli_result, boolean given in C:\xampp\htdocs\series\dynamic\AtomCMS\functions\data.php on line 8

**Comment by sami313** [comment]
<?php

function data_setting_vlaue($dbc, $id){
	
	$q = "SELECT * FROM settings WHERE id = '$id'";
	$r = mysqli_query($dbc, $q);

	$data = mysqli_fetch_assoc($r);
	
	return $data['value'];
}

function data_page($dbc, $id){
	
		$q = "SELECT * FROM pages WHERE id=$id";
		$r =  mysqli_query($dbc, $q);
			
		$data = mysqli_fetch_assoc($r);
		
		$data['body_nohtml'] =strip_tags($data['body']);
		
		if($data['body'] == $data['body_nohtml']){
			
			$data['body_formatted'] = '<p>'.$data['body'].'</p>';
			
		}else{
			
			$data['body_formatted'] =$data['body'];
			
		}
		
		
		return $data;
	
}

?>

**Comment by sami313** [comment]
function data_page is working fine ...as i wore the function data_setting_value it throught that error please help me :)

**Comment by JasonMate** [comment]
Is that a typo in function data_setting_vlaue($dbc, $id){}

**Comment by sami313** [comment]
ohhh sorry i got it thanks alot for helping out God bless you

**Comment by sami313** [comment]
function data_setting_value($dbc, $id){
		
	$q = "SELECT * FROM settings WHERE id = '$id'";
	$r = mysqli_query($dbc, $q);

	$data = mysqli_fetch_assoc($r);
  	
	return $data['value'];
 
} 
//function call in site page

#Site Setup

$debug = data_setting_value($dbc, 'debug-status');


i am still getting that error 
Warning: mysqli_fetch_assoc() expects parameter 1 to be mysqli_result, boolean given in C:\xampp\htdocs\series\dynamic\AtomCMS\functions\data.php on line 8

**Comment by creptor** [comment]
please I would be gratefull if you read this when placing code so I can help you better (-link-)[https://guides.github.com/features/mastering-markdown/#GitHub-flavored-markdown]

now I need you to check if you actually have a table named `settings` and that your table has a colum with the id equals to `debug-status`

**Comment by sami313** [comment]
yah sorry i will try to place the code that way i am new to github i will try to fallow the instructions

**Comment by sami313** [comment]
![localhost 127 0 0 1 atom settings phpmyadmin 4 2 11](https://cloud.githubusercontent.com/assets/11160392/6768579/921a9f3e-d096-11e4-9b64-e27d5fa00817.png)

**Comment by sami313** [comment]
![localhost 127 0 0 1 atom settings phpmyadmin 4 2 11 2](https://cloud.githubusercontent.com/assets/11160392/6768580/a5753a08-d096-11e4-96d9-f8610317965e.png)

**Comment by sami313** [comment]
![localhost 127 0 0 1 atom settings phpmyadmin 4 2 11 3](https://cloud.githubusercontent.com/assets/11160392/6768582/b34bbefe-d096-11e4-9e53-88cbfaf7ca1f.png)

**Comment by sami313** [comment]
i took the screen shot i hope it will help you to have a nice look thanks for your support :)

**Comment by creptor** [comment]
I have found no error on that code... are you sure that it's the code where line 8 is?

**Comment by creptor** [comment]
You could try with my prepared statements query, I know they work for me so.. maybe I couldn't see the error and using this code will fix it.

Code:
```
function data_setting_value($dbc, $id){
	$stmt = $dbc->query("SELECT * FROM settings WHERE id = '$id'");
	$data = $stmt->fetch_assoc();
	return $data['value'];
}
```

**Comment by sami313** [comment]
yah i am pretty sure should i delete that table and make it again  ?

    function data_setting_value($dbc, $id){
		
         	$q = "SELECT * FROM  settings WHERE id=$id";
	        $r = mysqli_query($dbc, $q) or die(mysql_error());

	        $data = mysqli_fetch_assoc($r);
  	
	         return $data['value'];
 }
* i tried to put that mqsql_error() function so it can give me whats is the error but when i had done this i just got white blank screen no error ...can you please tell me sql command which i can put directly in phpmyadmin to check it where there is something not wrong with sql command thanks :) *

**Comment by sami313** [comment]
###Fatal error: Call to a member function fetch_assoc() on boolean in C:\xampp\htdocs\series\dynamic\AtomCMS\functions\data.php on line 5### *i am getting this error after using the code you have told*

    function data_setting_value($dbc, $id){
        $stmt = $dbc->query("SELECT * FROM settings WHERE id = '$id'");
        $data = $stmt->fetch_assoc();
        return $data['value'];
}

**Comment by creptor** [comment]
the sql comend is what's inside the query.

I'm sorry i guess your php version does not support prepared statements.

**Comment by creptor** [comment]
this is the sql comand: "SELECT * FROM settings WHERE id = (place any id)"

**Comment by sami313** [comment]
*ok i will try to look in update phpversion*

**Comment by sami313** [comment]
![localhost 127 0 0 1 atom settings phpmyadmin 4 2 4 11](https://cloud.githubusercontent.com/assets/11160392/6769632/6f61ae80-d0c6-11e4-947f-bdc112c3ff55.png)

**Comment by creptor** [comment]
hey you had to change the (place any id) for an existing id

**Comment by sami313** [comment]
*i am sorry i think i might have gone goofy .. i didnt understand...i had put id = debug-status

**Comment by creptor** [comment]
Also i don't recoment you update the php version.... the problem is just why it's not working the code. I'll come back in like an ahour because i have to go now, but in the meanwile i'll try to figure out this problem.

**Comment by sami313** [comment]
*thanks for you great support me to i am also banging my head in the problem* :+1:

**Comment by creptor** [comment]
to use mysqli_erorr you have to put the dbc inside.... example:

`$r = mysqli_query($dbc, $q) or die(mysql_error($dbc));`

**Comment by creptor** [comment]
I guess to help you in this case i'll need the files. Could you please add them (the site) to like dropbox or something and share it with me.. my email is creptor@live.com

**Comment by sami313** [comment]
*i had made the table again with name set but same problem i almost pulled my half hairs of my head lol* yah thats really supportive of you i am just gone send you the code

**Comment by creptor** [comment]
I got it, when I arrive home i'll look it for any errors

**Comment by creptor** [comment]
hey I just noticed that in the files you gave me the function has this in is query: `$q = "SELECT * FROM set WHERE id = '$id'"` and not this: `$q = "SELECT * FROM settings WHERE id = '$id'"` ... look it up please

**Comment by sami313** [comment]
*i did that intentionly because i drop the settings table and make it as set so i thought it might work*

**Comment by sami313** [comment]
*hello any body there*

**Comment by creptor** [comment]
I was on school, I just came back :cry:

**Comment by creptor** [comment]
I'll try to recreate your error to see what's not working....

**Comment by creptor** [comment]
hey this works perfectly, the code is not the problem... it should work, I have try to recreated but it just don't send errors

code:
```
function data_setting_value($dbc, $id){
		
	$q = "SELECT * FROM settings WHERE id = '$id'";
	$r = mysqli_query($dbc, $q);
	
	$data = mysqli_fetch_assoc($r);
	
	return $data['value'];
 
} 
```

check your $debug, it should be like this:
```
$debug = data_setting_value($dbc, 'debug_status');
```

**Comment by sami313** [comment]
*i feel there is something wrong with my database i wonder what have gone wrong so far in code i dont even my self find any error and you have a look as well*

**Comment by JasonMate** [comment]
In the screenshot you provided, It looks like there is an space included on your the name of your settings table. Take a look: ` settings`

**Comment by sami313** [comment]
*i even drop the table settings and made a new one with name set but my error is still the same*  :(

**Comment by JasonMate** [comment]
What does your setup.php look like?
Do you have a line that looks something like this?
```
$debug = data_setting_value($dbc, 'debug-status');
```

**Comment by sami313** [comment]
![untitled](https://cloud.githubusercontent.com/assets/11160392/6850471/3879f64c-d3fb-11e4-9573-85165d3d7877.png)

**Comment by JasonMate** [comment]
I dont see anything wrong with that file, but I would remove that comment
off github because it's showing your database information.

**Comment by sami313** [comment]
*yah i will remove it as well point was that to figure the problem *

**Comment by sami313** [comment]
*can any of you guy can look remotely into my problem ?*

**Comment by creptor** [comment]
to look remotelly you have to be connected when whe're but be carefull I'll not harm in any way your computer but do not give your ip to any stranger(facebook, youtube, etc)

**Comment by creptor** [comment]
@JasonMate you're right....  It seems the table that you have created has spaces inside of the name! that's wy is not working.

**Comment by sami313** [comment]
*that settting table i have drop that long ago i made that with set name i thought that if i had some problem there i might get rid of it and make new table with set name @creptor can you please tell me which software you use to connect so i can install it ..i heard about teamviewer if you have any other option please tell me*

**Comment by creptor** [comment]
i use the web software preinstalled with xampp or wamp wich is php my admin :3 - and sql,but it's basically the same thing.

**Comment by sami313** [comment]
* i have no idea on that can you give me that name so i can install it as well and you can look in my problem remotely*

**Comment by creptor** [comment]
to continue this with remote connection email me at creptor@live.com (time: UTC−03:00) with your ip, and i'll contact you if you have any errors...by mail, so keep your router UP(ON).
